### PR TITLE
[15.0][MIG] sale_coupon_limit: Migration to 15.0

### PR DESCRIPTION
**Pull request by chienandalu**
- [x] Partially superseeds #33 although adapting it to v15.0 to respect changes attribution
- [x] Additional adaptations
- [x] Depends on #94 
- [x] Promote to stable sale_commercial_partner https://github.com/OCA/sale-workflow/pull/2342

cc @Tecnativa TT38310

please review @pedrobaeza @victoralmau

### CODE REVIEW

**Comment by pedrobaeza** [comment]
This must be renamed to `coupon_limit`, isn't it?

**Comment by chienandalu** [comment]
> This must be renamed to `coupon_limit`, isn't it?

It depends on features from sale_coupon

**Comment by pedrobaeza** [comment]
But should we split it or make it less dependant?

**Review by victoralmau — APPROVED** [review]
Code review OK

**Comment by pedrobaeza** [comment]
This should include https://github.com/OCA/sale-promotion/pull/76/files#diff-c3be12f82d41fbacd8bf07d5ee671f0e364f35845ddb8ab6dd20d165fb7fce00

File: sale_coupon_limit/models/coupon_coupon.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            ("program_id", "=", self.program_id.id),
+            ("state", "=", "used"),
+        ]
+        # Customer limit rules
+        if self.program_id.rule_max_customer_application:
+            coupons_count = self.search_count(
+                domain
+                + [
+                    (
+                        "sales_order_id.commercial_partner_id",
```
Comment (pedrobaeza) [inline sale_coupon_limit/models/coupon_coupon.py]: The coupon may be applied to other things like a PoS order, so you should try to change this code to be more generic (isn't there a count in coupon itself for all?) and if possible, rename the module without `sale_` prefix.

**Comment by chienandalu** [comment]
I don't know how I got to close this. Can you reopen?

File: sale_coupon_limit/models/coupon_coupon.py (hunk 1 above)
Comment (chienandalu) [reply sale_coupon_limit/models/coupon_coupon.py]: This method is exclusive of `sales_coupon`. PoS coupons are checked in a separate js logic

**Comment by pedrobaeza** [comment]
I can't open it I'm afraid

**Pull request closed without merging**